python/cudf/cudf/window/rolling.py

```diff
@@ -164,6 +164,9 @@ def mean(self):
     def count(self):
         return self._apply_agg("count")
 
+    def udf(self, op):
```

Comment:
I think we want to expose the UDF functionality differently here.

1. Rather than `rolling.udf(op)`, how about `rolling.apply(ptx, *args, **kwargs)`? This can later be extended to `rolling.apply(ptx_or_func, *args, **kwargs)` - a more familiar API to Pandas users.

2. Rather than passing a `tuple`, we can require that `dtype` be specified in `**kwargs` if the first argument is PTX.

Reply:
@shwina How do we know if the first argument is PTX or not? My python knowledge is very limited >_< Does this mean we require `output_type` is specified in `**kwargs`?

Reply:
Since PTX is passed in as a string, there's no way to know for sure. Does `libcudf` have a utility to check if a string represents valid PTX code? If so, we could write a Cython wrapper around it that raises a `ValueError` when the PTX is not valid. So overall, something like:

```python
def apply(ptx, *args, **kwargs):
    check_valid_ptx(ptx)
    if "dtype" not in kwargs:
        raise TypeError("Missing keyword argument 'dtype=' in apply")
    return self._apply_agg((ptx, dtype))
```